_posts/2024-08-07-taming-apache-kafka-4.0.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: post
+title:  "Taming Apache Kafka 4.0"
+date: 2024-08-07
+author: jakub_scholz
+---
+
+Apache Kafka 4.0 release is just around the corner.
+It will be a very significant release for Apache Kafka.
+But it will also have a huge impact on Strimzi.
+Therefore in this blog post, we will look closer at what changes will Apache Kafka 4.0 bring, when is it currently planned to be released and how will it be supported by Strimzi.
+
+<!--more-->
+
+### Why is Apache Kafka 4.0 so impactful for Strimzi
+
+New major releases often bring a lot of changes, remove old deprecated APIs, and so on.
+And Apache Kafka 4.0 won't be different.
+But unlike the previous Apache Kafka major releases that Strimzi went through and which had only minimal impact on us, Kafka 4.0 will have also some changes with a huge impact on Strimzi:
+
+* Kafka 4.0 will not support ZooKeeper-based Apache Kafka clusters anymore.
+  The only way how to run a Kafka cluster will be by using KRaft.
+  That means that all existing ZooKeeper-based clusters will need to be migrated to KRaft before upgrading to Apache Kafka 4.0.
+* Mirror Maker 1 was already deprecated in Kafka 3.0 and it will be completely removed in Kafka 4.0.
+  You will need to migrate to Mirror Maker 2 or some other mirroring tool of your choice before upgrading to Kafka 4.0.
```

Comment:
```suggestion
* MirrorMaker 1 was already deprecated in Kafka 3.0 and it will be completely removed in Kafka 4.0.
  You will need to migrate to MirrorMaker 2 or some other mirroring tool of your choice before upgrading to Kafka 4.0.
```